### DISCUSSION TITLE
Where set zigbee polling / reporting intervall?

### DISCUSSION THREAD

**Discussion by ghost** [arendst/Tasmota#21937] [Q&A]
Is there any command to set the zigbee polling interval? Or where i can find the interval in the source code, so i would compile my own version?

I have a zigbee device, which is reporting its status every 150 to 160 seconds. In an older version of tasmota it was a fixed 180 seconds interval. So i think its hardcoded somewhere in the Z2T code?
I plan to install many more zigbee devices, so I need to reduce the traffic in the zigbee network.

Thank you!

What i tried but without success:

SetOption 110 1
SetOption 116 1
teleperiod 0

BTW: With Z2M over a serial interface to the cc2530 i don't receive these messages.

```
04:52:33.385 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":60}}}
04:55:16.277 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":60}}}
04:58:04.688 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":60}}}
05:00:45.642 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:03:22.442 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":76}}}
05:05:54.514 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:08:25.304 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":76}}}
05:11:02.766 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:13:55.414 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":68}}}
05:16:32.692 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":70}}}
05:19:20.919 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":65}}}
05:21:56.213 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":73}}}
05:24:44.743 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:27:21.919 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":65}}}
05:30:00.930 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:32:55.624 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":55}}}
```

Detailed Log of such a status message:

```
05:43:40.315 ZIG: {"ZbZNPReceived":"45C4D29D00"}
05:43:40.367 ZIG: {"ZbZNPReceived":"448100000000D29D0101003F0098CBB600000F08B60A01002042E2FF2036E4FF2000D29D1D"}
05:43:40.369 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0000","srcaddr":"0x9DD2","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":63,"securityuse":0,"seqnumber":0,"fc":"0x08","frametype":0,"direction":1,"disableresp":0,"manuf":"0x0000","transact":182,"cmdid":"0x0A","payload":"01002042E2FF2036E4FF2000"}}
05:43:40.372 ZigbeeZCLSend device: 0x9DD2, endpoint:1, cluster:0x0000, cmd:0x0B, send:"0A00"
05:43:40.374 ZIG: ZbZNPSent 240202D29D000000000000010000010000B6001E050010B60B0A00
05:43:40.376 ZIG: ZbZCLRawReceived: {"0x9DD2":{"0000/0001":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}
05:43:40.380 RSL: SENSOR = {"ZbReceived":{"0x9DD2":{"Device":"0x9DD2","AppVersion":66,"0000/FFE2":54,"0000/FFE4":0,"Endpoint":1,"LinkQuality":63}}}
05:43:40.415 ZIG: {"ZbZNPReceived":"640200"}
05:43:40.465 ZIG: {"ZbZNPReceived":"44800001B6"}
```

**Comment by s-hadinger** [comment]
There is no polling in zigbee. This is not the way it works. It's the end device that decides the frequency of reporting for attributes.

**Reply by sfromis** [reply]
Depending on device, you may (or not) be able to configure the reporting interval, example:
```
ZbSend {"Device":"Ikea-Vindstyrka","Config":{"Humidity":{"MinInterval":300,"MaxInterval":3600,"ReportableChange":1}}}
```
Tasmota only reports what it gets, and it is not related to `TelePeriod`.

**Reply by ghost** [reply]
> There is no polling in zigbee. This is not the way it works. It's the end device that decides the frequency of reporting for attributes.

And why then in zigbee2mqtt over HA I don't see this reports - even in debug mode?
Then in tried IoBroker (zigbee plugin) - there i see the reports every 5 minutes.
As mentioned in an older version of Tasmota the inverval was exactly 3 minutes.

I know the device driven reports - but I think that the above messages are somehow initated by tasmota.

**Reply by ghost** [reply]
> Depending on device, you may (or not) be able to configure the reporting interval, example:
> 
> ```
> ZbSend {"Device":"Ikea-Vindstyrka","Config":{"Humidity":{"MinInterval":300,"MaxInterval":3600,"ReportableChange":1}}}
> ```
> 
> Tasmota only reports what it gets, and it is not related to `TelePeriod`.

Yes, i know this possibility to set the device-driven reports. But in my case, I don't know in which cluster and attribute i should write the config.

**Reply by sfromis** [reply]
When the clusters/attributes were not decoded, I'd not assume it to be the settings Z2T may be sending during auto-bind after pairing, but you could check what happens on the log when pairing again.

**Reply by s-hadinger** [reply]
Those attributes are obviously not zigbee standard. It can happen that the device expects a "magic spell" message to stop sending reports. Or you maybe just not doing the auto binding would work as well. 

But I can assure you that Z2T does not do polling. The logs you sent had no Z2T initiated message
